lib/wasm/WebAssemblyGenerator.js

```diff
@@ -189,19 +189,86 @@ const rewriteImportedGlobals = state => bin => {
 
 				newGlobals.push(
 					t.global(globalType, [
-						t.objectInstruction("const", "i32", [t.numberLiteralFromRaw(0)])
+						// Poisong globals, they are meant to be rewritten
+						t.objectInstruction("const", "i32", [t.numberLiteralFromRaw(666)])
 					])
 				);
 
 				path.remove();
 			}
+		},
+
+		// in order to preserve non-imported global's order we need to re-inject
+		// those as well
+		Global(path) {
+			newGlobals.push(path.node);
+			path.remove();
```

Comment:
That fixes an unrelated bug

Reply:
Do you want to send this as separate PR?

Reply:
Yes ok, I can add tests too.